### DISCUSSION TITLE
Is it possible to inbuild Zygisk in kernel ?

### DISCUSSION THREAD

**Discussion by ghost** [tiann/KernelSU#1291] [Ideas]
I'm noob and new into Linux kernel, and I want to know if it is possible to build Zygisk (or another hooking implementation) into kernel without using a module. This should be undetectable.

I did all super subpartitions read-write using a script and modify framework.jar including [Pine](https://github.com/canyie/pine) to spoof a "valid" certificate chain, I hooked a lot of methods and apps like Momo didn't detect it. Imagine this in kernel...

How viable would this be?

**Comment by backslashxx** [comment]
isnt zygisk a thing that injects to a zygote process?
so its on userspace

**Comment by androidacy-user** [comment]
Zygote is inherently userspace. The sheer amount of hacks to try to hook it from the kernel is unimaginable.

[a comment by an automated account is not shown]
